parse_snyk_output.cjs

```diff
@@ -0,0 +1,178 @@
+
+  // JQL query to check ticket given title from the past given days
+  let jqlQuery = `project = "${process.env.JIRA_PROJECT_KEY}" AND summary ~ "MCR - ${vulnerability.title}" AND created >= ${sixtyDaysAgo}`;
+  let searchResult = await jira.searchJira(jqlQuery);
+
+
+  if (searchResult.issues && searchResult.issues.length > 0) {
+    for (const issue of searchResult.issues) {
+      if (issue.fields.status.name !== "Closed" && issue.fields.status.name !== "Cancelled") {
+        if (issue.fields.summary.startsWith('[MCR] -')) {
+          console.log(`Active Jira ticket already exists for vulnerability: ${vulnerability.title}`);
+          return;
+        }
+      }
+      
+    }
+  } else {
+                const issue = {
+                    fields: {
+                      project: {
+                        key: process.env.JIRA_PROJECT_KEY,
+                      },
+                      summary: `${process.env.JIRA_TITLE_PREFIX}  ${vulnerability.title}`,
+                      description: vulnerability.description,
+                      issuetype: {
+                        name: process.env.JIRA_ISSUE_TYPE,
+
+                      },
+                      labels: process.env.JIRA_LABELS.split(','),
+                      "customfield_10007" : process.env.JIRA_EPIC_KEY,
+                    },
+                  };
+
+                  
+
+                  let issueResponse;
+
+                  try {
+                    const jiraUrl = `${process.env.JIRA_BASE_URL}/rest/api/2/issue`;
+                    console.log('JIRA_URL:', jiraUrl);
+                    issueResponse = await jira.addNewIssue(issue);
+                    console.log(issueResponse)
+                    console.log(`Jira ticket created for vulnerability: ${vulnerability.title}`);
+                  } catch (error) {
+                    console.error('Error creating Jira ticket:', error);
+                  }
+
+                  // attach the scan report to the Jira ticket
+                  const attachmentUrl = `https://${process.env.JIRA_BASE_URL}/rest/api/2/issue/${issueResponse.key}/attachments`;
+                  const attachment = {
+                    method: 'POST',
+                    uri: attachmentUrl,
+                    auth: {
+                      username: process.env.JIRA_USER_EMAIL,
+                      password: process.env.JIRA_API_TOKEN
+                    },
+
+                    headers: {
+                      'X-Atlassian-Token': 'no-check', // Disable XSRF check for file upload
+                      'X-Requested-With': 'XMLHttpRequest',
+                      'X-Scheme': 'https',
+                      'X-Forwarded-Proto': 'https',
+                    },
+
+                    formData: {
+                      file: fs.createReadStream('snyk_output.txt')
+                    },
+                  };
+                  
+                  try {
+                    await rp(attachment);
```

Comment:
Blocking: The `request-promise` library was [deprecated three years ago](https://www.npmjs.com/package/request-promise). If we can use the jira API directly, then we won't need `rp` at all. Otherwise, we should use an alternative library. `axios` and `node-fetch` seem promising. In fact it looks like you already tried the latter?

Reply:
Addressed!